src/components/bootcamp-interns/BootcampInternEditForm.tsx

```diff
@@ -0,0 +1,142 @@
+import * as yup from 'yup'
+import { AxiosError } from 'axios'
+import { observer } from 'mobx-react'
+import { FormikHelpers } from 'formik'
+import { useRouter } from 'next/router'
+import { makeStyles } from '@mui/styles'
+import { useTranslation } from 'next-i18next'
+import { UseBaseQueryResult, useMutation } from 'react-query'
+import { ButtonGroup, Grid, Typography, Button } from '@mui/material'
+
+import { routes } from 'common/routes'
+import { axios } from 'common/api-client'
+import { endpoints } from 'common/api-endpoints'
+import { name, email } from 'common/form/validation'
+import GenericForm from 'components/common/form/GenericForm'
+import { BootcampIntern } from 'lib/interfaces/BootcampIntern'
+import SubmitButton from 'components/common/form/SubmitButton'
+import FormTextField from 'components/common/form/FormTextField'
+import { useFetchBootcampIntern } from 'common/hooks/bootcampIntern'
+import { BootcampInternInput, BootcampInternResponse } from 'gql/bootcamp'
+import { ApiErrors, isAxiosError, matchValidator } from 'common/api-errors'
+import { NotificationStore } from 'stores/bootcamp-interns/NotificationStore'
+
+import { drawerWidth } from './MyDrawer'
+
+const useStyles = makeStyles(() => {
+  return {
+    internForm: {
+      marginLeft: drawerWidth,
+      marginTop: '200px',
+    },
+    internFormHeader: {
+      marginBottom: '50px',
+    },
+  }
+})
+
+const validationSchema: yup.SchemaOf<BootcampInternInput> = yup.object().shape({
+  firstName: name,
+  lastName: name,
+  email: email,
+  id: yup.string(),
+})
+
+export default observer(function BootcampInternEditForm() {
+  const { setNotificationMessage, showNotification } = NotificationStore
+
+  const router = useRouter()
+  const { t } = useTranslation()
+  const classes = useStyles()
+  const internId = router.query.id
+
+  if (typeof internId !== 'string') {
+    showNotification()
+    setNotificationMessage('Invalid id passed , please try again.')
+    router.push(routes.bootcampIntern.index)
+    return null
+  }
+
+  const { data, isLoading }: UseBaseQueryResult<BootcampIntern> = useFetchBootcampIntern(internId)
+
+  const defaults: BootcampInternResponse = {
+    firstName: data?.firstName,
+    lastName: data?.lastName,
+    email: data?.email,
+  }
+
+  const submitIntern = async (internData: BootcampIntern) => {
+    await axios.patch(endpoints.bootcampIntern.listBootcampIntern.url + `/${internId}`, internData)
```

Comment:
You might improve this further by defining a dedicated API endpoint with the patch action and typed function to accept the `internId` param

```tsx
updateIntern: (id: string) => <Endpoint>{ url: `/bootcamp-intern/${id}`, method: 'PATCH' },
```

Every API endpoint should have well defined and documented definition stored in a single location. This is much clearer than having string concatenation here and there.

```
endpoints.bootcampIntern.listBootcampIntern.url + `/${internId}`
```

Reply:
Yeah that looks better and logical , thanks.